cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -46,23 +46,15 @@ const (
 	resourceNamePrefix   = "intel.com/fpga"
 )
 
-type pluginMode int
+type pluginMode string
```

Comment:
Consider the function
```
func foo(bar string, mode string) {
...
}
```
In this form it's error prone, because a user may mess up the arguments and the compiler won't prevent it from happening:
```
foo(myMode, myBar)
```

With this definition:
```
func foo(bar string, mode pluginMode){
...
}
```
the compiler will prevent wrong argument order.